src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4553,10 +4553,24 @@ internal static IEnumerable<CompletionResult> CompleteFilename(CompletionContext
                 basePath = EscapePath(basePath, stringType, useLiteralPath, out _);
             }
 
-            _ = context.Helper
+            PowerShell currentPS = context.Helper
                 .AddCommandWithPreferenceSetting("Microsoft.PowerShell.Management\\Resolve-Path")
                 .AddParameter("Path", basePath);
 
+            string relativeBasePath;
+            var lastAst = context.RelatedAsts[^1];
+            if (lastAst.Parent is UsingStatementAst usingStatement
+                && usingStatement.UsingStatementKind is UsingStatementKind.Module or UsingStatementKind.Assembly
+                && lastAst.Extent.File is not null)
+            {
+                relativeBasePath = Directory.GetParent(lastAst.Extent.File).FullName;
+                _ = currentPS.AddParameter("RelativeBasePath", relativeBasePath);
```

Comment:
What if basePath is absolute path?

Reply:
There's no direct relationship between `basePath` and `relativeBasePath` so it doesn't matter what format the path is in `basePath`. But if it is an absolute path `useRelativePath` will most likely be false and because of that `relativeBasePath` won't be used for anything.

Reply:
Then this code would have to be under `if (useRelativePath)`?

Reply:
The method `GetFileSystemProviderResults` still expects a value for the `relativeBasePath` parameter so I don't see any reason to do that. It would just be `if (useRelativePath) {} else {`relativeBasePath` = string.Empty}`

Reply:
This would make the code more understandable.
(As a separate note, when I see this code (completion as whole) I always feel like it's too confusing and needs to be redesigned.)

Reply:
I'm not sure I agree that it's better, but I've added it with: https://github.com/PowerShell/PowerShell/pull/20017/commits/f7b3e4379df07bf269be1bf8729916f7b1b3527a
As for the side note, I agree it could be better but it's not so bad that I think a complete rewrite is worth the effort. Code completion generally works pretty well in PowerShell and if I see an issue I can fairly quickly find the problematic code.